crates/vizia_core/src/systems/text_constraints.rs

```diff
@@ -36,112 +34,62 @@ pub fn text_constraints_system(cx: &mut Context, tree: &Tree<Entity>) {
             }
 
-            // set align/baseline and move the start coordinate to the appropriate place in the box
-            let align = match (child_left, child_right) {
-                (Units::Stretch(_), Units::Stretch(_)) => {
-                    x += w / 2.0;
-                    Align::Center
-                }
-                (Units::Stretch(_), _) => {
-                    x += w;
-                    Align::Right
-                }
-                _ => Align::Left,
-            };
-            let baseline = match (child_top, child_bottom) {
-                (Units::Stretch(_), Units::Stretch(_)) => {
-                    y += h / 2.0;
-                    Baseline::Middle
-                }
-                (Units::Stretch(_), _) => {
-                    y += h;
-                    Baseline::Bottom
-                }
-                _ => Baseline::Top,
-            };
-
             let mut content_width = 0.0;
             let mut content_height = 0.0;
 
-            if let Some(text) = cx.style.text.get(entity).cloned() {
-                let mut paint = text_paint_general(&cx.style, &cx.resource_manager, entity);
-                paint.set_text_align(align);
-                paint.set_text_baseline(baseline);
-
-                let font_metrics =
-                    cx.text_context.measure_font(&paint).expect("Failed to read font metrics");
-
-                if let Ok(lines) = text_layout(f32::MAX, &text, &paint, &cx.text_context) {
-                    let metrics = measure_text_lines(&text, &paint, &lines, x, y, &cx.text_context);
-                    let text_width = metrics
-                        .iter()
-                        .map(|m| m.width())
-                        .reduce(|a, b| a.max(b))
+            if cx.cosmic_context.has_buffer(entity) {
+                cx.cosmic_context.sync_styles(entity, &cx.style);
+                let (text_width, text_height) = cx.cosmic_context.with_buffer(entity, |buf| {
+                    buf.set_size(i32::MAX, i32::MAX);
+                    let w = buf
+                        .layout_runs()
+                        .filter_map(|r| (!r.line_w.is_nan()).then_some(r.line_w))
+                        .max_by(|f1, f2| f1.partial_cmp(f2).unwrap())
                         .unwrap_or_default();
-                    let text_height = font_metrics.height().round() * metrics.len() as f32;
+                    let h = buf.layout_runs().len() as f32 * buf.metrics().line_height as f32;
+                    (w, h)
+                });
 
-                    // Add an extra pixel to account for AA
-                    let text_width = text_width.round() + 1.0 + child_space_x;
-                    let text_height = text_height.round() + 1.0 + child_space_y;
+                // Add an extra pixel to account for AA
+                let text_width = text_width.round() + 1.0 + child_space_x;
+                let text_height = text_height.round() + 1.0 + child_space_y;
```

Comment:
So before I added the extra 1px padding for AA because of how femtovg does AA (it adds a 1px edge). I'm wondering if we need this any more?